scipy/stats/_continuous_distns.py

```diff
@@ -4948,7 +4948,15 @@ class loggamma_gen(rv_continuous):
 
     """
     def _rvs(self, c, size=None, random_state=None):
-        return np.log(random_state.gamma(c, size=size))
```

Comment:
one could also consider using the direct approach for c larger than some lower bound and the new approach for small c (geninvgauss is a distribution that uses different methods for different parameter values). 

pro: 
- a bit faster when log(Gamma(...)) is used
- the rvs returned do not change, so no need to save the data

con:
- implementation of rvs is a bit longer

what is your view?

Reply:
Very reasonable, but array `c` needs to be considered, and would take some testing to determine whether it's worth the overhead of `_lazywhere` or some other approach. Let's save performance recovery for a separate PR, since this makes an important accuracy improvement at moderate performance cost.